pyartcd/pyartcd/pipelines/ocp4.py

```diff
@@ -0,0 +1,96 @@
+import os
+
+import click
+from aioredlock import LockError
+
+from pyartcd import exectools, locks
+from pyartcd.cli import cli, pass_runtime, click_coroutine
+from pyartcd.runtime import Runtime
+
+
+async def sync_repo_to_s3_mirror(local_dir: str, s3_path: str, dry_run: bool):
+    if s3_path.startswith('/pub/openshift-v4/clients') or \
+            s3_path.startswith('/pub/openshift-v4/amd64') or \
+            s3_path.startswith('/pub/openshift-v4/arm64') or \
+            s3_path.startswith('/pub/openshift-v4/dependencies'):
+        raise Exception(
+            f'Invalid location on s3 ({s3_path}); these are virtual/read-only locations on the s3 '
+            'backed mirror. Qualify your path with /pub/openshift-v4/<brew_arch_name>/ instead.')
+
+    # Sync is not transactional. If we update repomd.xml before files it references are populated,
+    # users of the repo will get a 404. So we run in three passes:
+    # 1. On the first pass, exclude files like repomd.xml and do not delete any old files.
+    #    This ensures that we  are only adding new rpms, filelist archives, etc.
+    # 2. On the second pass, include only the repomd.xml.
+    base_cmd = ['aws', 's3', 'sync', '--no-progress', '--exact-timestamps']
+    if dry_run:
+        base_cmd.append('--dryrun')
+
+    cmd = base_cmd + [
+        '--exclude', '*/repomd.xml', local_dir,
+        f's3://art-srv-enterprise{s3_path}'  # Note that s3_path has / prefix.
+    ]
+    env = os.environ.copy()
+    await exectools.cmd_assert_async(cmd, env=env)
+
+    cmd = base_cmd + [
+        '--exclude', '*', '--include', '*/repomd.xml', local_dir,
+        f's3://art-srv-enterprise{s3_path}'
+    ]
+    await exectools.cmd_assert_async(cmd, env=env)
```

Comment:
there will be a potential repomd.xml updated before previous cmd run complete

Reply:
The two `aws s3 sync` operations are sequential. We will `await` for the first one to complete before moving on to the next one.